include/deal.II/base/index_set.h

```diff
@@ -229,6 +229,12 @@ class IndexSet
   void compress () const;
 
   /**
+   * Reorder the ranges inside IndexSet so that the largest range is the first
+   * one.
+   */
+  void reorder_ranges();
```

Comment:
Can you do this? Do none of the member functions of this class ever expect the ranges to be sorted in ascending order?

Maybe a better name for the function would be `optimize_range_order()` or similar. Or we could simply make this the default and hide it in `compress()`.